inject.js

```diff
@@ -41,35 +42,47 @@ function wrap (fn) {
   }
 }
 
-module.exports = function inject (blobs, set, name, opts) {
+module.exports = function inject (blobStore, blobPush, name, opts) {
   opts = opts || {}
   // sympathy controls whether you'll replicate
   const sympathy = opts.sympathy == null ? 3 : opts.sympathy | 0
   const stingy = opts.stingy === true
   const legacy = opts.legacy !== false
   const pushy = opts.pushy || 3
-  const max = opts.max || 5 * 1024 * 1024
+  const max = opts.max || MAX_SIZE
+
+  const peers = {} // currently connected peers - { Peer: RPC }
+  const peerBlobs = {} // tracks blob locations - { Peer: { BlobId: Size } }
+  const streams = {}
 
-  const notify = Notify()
-  const pushed = Notify()
+  const want = {}
+  const push = {}
 
-  const peers = {}
-  const want = {}; const push = {}; const waiting = {}; const getting = {}
-  const available = {}; const streams = {}
-  let send = {}
+  const waiting = {}
+  const getting = {}
+
+  const pushed = Notify() // makes a stream of blob hashes which have reached 'pushy' goal
+  const notify = Notify() // used with createWants + legacySync
+
+  blobPush.updates(update => {
+    const notification = {}
+    Object.keys(update).forEach(blobId => {
+      notification[blobId] = -1
+      push[blobId] = push[blobId] || {}
+    })
+    notify(notification) // update all currently connected peers about new push
+  })
```

Comment:
this `blobPush.updates` listener is one important change in this file  - ensures that even if stream to a peer has already been opened, any new pushes will for sure make it to them